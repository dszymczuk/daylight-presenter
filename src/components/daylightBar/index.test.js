import React from "react";
import { render, cleanup } from '@testing-library/react';
import DaylightAcrossDays from "./daylightAcrossDays";
import DaylightDuringDay from "./daylightDuringDay";
import Daylight from "./index";

afterEach(cleanup);

it("renders DaylightAcrossDays", () => {
  const props = {
    sunrise: "6:23:54 AM",
    sunset: "7:12:32 PM"
  };
  const { asFragment } = render(<DaylightAcrossDays {...props} />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders DaylightDuringDay", () => {
  const props = {
    sunrise: "6:23:54 AM",
    sunset: "7:12:32 PM"
  };
  const { asFragment } = render(<DaylightDuringDay {...props} />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders Daylight when sunset is after sunrise", () => {
  const daylight = {
    sunrise: "6:23:54 AM",
    sunset: "7:12:32 PM"
  };
  const { asFragment } = render(<Daylight daylight={daylight}/>);
  expect(asFragment()).toMatchSnapshot();
});

it("renders Daylight when sunset is after sunrise", () => {
  const daylight = {
    sunrise: "6:23:54 PM",
    sunset: "7:12:32 AM"
  };
  const { asFragment } = render(<Daylight daylight={daylight}/>);
  expect(asFragment()).toMatchSnapshot();
});