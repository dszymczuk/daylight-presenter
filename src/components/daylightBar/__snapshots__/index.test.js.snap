// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Daylight when sunset is after sunrise 1`] = `
<DocumentFragment>
  <div
    class="progressBarWrapper"
  >
    <div
      class="progress"
    >
      <div
        aria-valuemax="1440"
        aria-valuemin="0"
        aria-valuenow="383"
        class="progress-bar bg-success"
        role="progressbar"
        style="width: 26.597%;"
      />
      <div
        aria-valuemax="1440"
        aria-valuemin="0"
        aria-valuenow="768"
        class="progress-bar bg-warning"
        role="progressbar"
        style="width: 53.333%;"
      />
      <div
        aria-valuemax="1440"
        aria-valuemin="0"
        aria-valuenow="289"
        class="progress-bar bg-success"
        role="progressbar"
        style="width: 20.069%;"
      />
    </div>
    <div
      class="labels"
    >
      <div
        class="startHour"
      >
        00:00
      </div>
      <div
        class="endHour"
      >
        23:59
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders Daylight when sunset is after sunrise 2`] = `
<DocumentFragment>
  <div
    class="progressBarWrapper"
  >
    <div
      class="progress"
    >
      <div
        aria-valuemax="1440"
        aria-valuemin="0"
        aria-valuenow="432"
        class="progress-bar bg-warning"
        role="progressbar"
        style="width: 30%;"
      />
      <div
        aria-valuemax="1440"
        aria-valuemin="0"
        aria-valuenow="671"
        class="progress-bar bg-success"
        role="progressbar"
        style="width: 46.597%;"
      />
      <div
        aria-valuemax="1440"
        aria-valuemin="0"
        aria-valuenow="335"
        class="progress-bar bg-warning"
        role="progressbar"
        style="width: 23.264%;"
      />
    </div>
    <div
      class="labels"
    >
      <div
        class="startHour"
      >
        00:00
      </div>
      <div
        class="endHour"
      >
        23:59
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders DaylightAcrossDays 1`] = `
<DocumentFragment>
  <div
    class="progressBarWrapper"
  >
    <div
      class="progress"
    >
      <div
        aria-valuemax="1440"
        aria-valuemin="0"
        aria-valuenow="1152"
        class="progress-bar bg-warning"
        role="progressbar"
        style="width: 80%;"
      />
      <div
        aria-valuemax="1440"
        aria-valuemin="0"
        aria-valuenow="769"
        class="progress-bar bg-success"
        role="progressbar"
        style="width: 53.403%;"
      />
      <div
        aria-valuemax="1440"
        aria-valuemin="0"
        aria-valuenow="1055"
        class="progress-bar bg-warning"
        role="progressbar"
        style="width: 73.264%;"
      />
    </div>
    <div
      class="labels"
    >
      <div
        class="startHour"
      >
        00:00
      </div>
      <div
        class="endHour"
      >
        23:59
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders DaylightDuringDay 1`] = `
<DocumentFragment>
  <div
    class="progressBarWrapper"
  >
    <div
      class="progress"
    >
      <div
        aria-valuemax="1440"
        aria-valuemin="0"
        aria-valuenow="383"
        class="progress-bar bg-success"
        role="progressbar"
        style="width: 26.597%;"
      />
      <div
        aria-valuemax="1440"
        aria-valuemin="0"
        aria-valuenow="768"
        class="progress-bar bg-warning"
        role="progressbar"
        style="width: 53.333%;"
      />
      <div
        aria-valuemax="1440"
        aria-valuemin="0"
        aria-valuenow="289"
        class="progress-bar bg-success"
        role="progressbar"
        style="width: 20.069%;"
      />
    </div>
    <div
      class="labels"
    >
      <div
        class="startHour"
      >
        00:00
      </div>
      <div
        class="endHour"
      >
        23:59
      </div>
    </div>
  </div>
</DocumentFragment>
`;
